src/Collector.cs

```diff
@@ -127,19 +142,30 @@ private void SendQueuedMessages(Socket socket)
         }
     }
 
-    private void Authenticate(Socket socket)
+    private bool Authenticate(Socket socket)
     {
-      var data = System.Text.Encoding.ASCII.GetBytes("hello version 1.0\n");
+      var data = System.Text.Encoding.ASCII.GetBytes("hello version dotnet/0.2.0\n");
       socket.Send(data);
-      data = System.Text.Encoding.ASCII.GetBytes(String.Format("authenticate {0}\n", _apiKey));
+      if(!ReceiveOk(socket)) return false;
```

Comment:
As a speed-up for handshake you can send `hello` and `authenticate` and then check for 2 `ok` responses. This eliminates a single round trip and still maintains the protocol.